lib/OptimizationManopt/src/OptimizationManopt.jl

```diff
@@ -0,0 +1,307 @@
+
+function call_manopt_optimizer(opt::QuasiNewtonOptimizer{Teval},
+        loss,
+        gradF,
+        x0,
+        stopping_criterion::Union{Nothing, Manopt.StoppingCriterion}) where {
+        Teval <:
+        AbstractEvaluationType
+}
+    sckwarg = stopping_criterion_to_kwarg(stopping_criterion)
+    opts = quasi_Newton(opt.M,
+        loss,
+        gradF,
+        x0;
+        return_state = true,
+        evaluation = Teval(),
+        retraction_method = opt.retraction_method,
+        vector_transport_method = opt.vector_transport_method,
+        stepsize = opt.stepsize,
+        sckwarg...)
+    # we unwrap DebugOptions here
+    minimizer = Manopt.get_solver_result(opts)
+    return (; minimizer = minimizer, minimum = loss(opt.M, minimizer), options = opts),
+    :who_knows
+end
+
+## Optimization.jl stuff
+
+function build_loss(f::OptimizationFunction, prob)
+    function (::AbstractManifold, θ)
+        x = f.f(θ)
+        __x = first(x)
+        return prob.sense === Optimization.MaxSense ? -__x : __x
+    end
+end
+
+function build_gradF(f::OptimizationFunction{true}, prob, cur)
+    function g(M::AbstractManifold, G, θ)
+        f.grad(G, θ, cur...)
+        G .= riemannian_gradient(M, θ, G)
+    end
+    function g(M::AbstractManifold, θ)
+        G = zero(θ)
+        f.grad(G, θ, cur...)
+        return riemannian_gradient(M, θ, G)
+    end
+end
+
+# TODO:
+# 1) convert tolerances and other stopping criteria
+# 2) return convergence information
+# 3) add callbacks to Manopt.jl
+
+function SciMLBase.__solve(prob::OptimizationProblem,
+        opt::AbstractManoptOptimizer,
+        data = Optimization.DEFAULT_DATA;
+        callback = (args...) -> (false),
+        maxiters::Union{Number, Nothing} = nothing,
+        maxtime::Union{Number, Nothing} = nothing,
+        abstol::Union{Number, Nothing} = nothing,
+        reltol::Union{Number, Nothing} = nothing,
+        progress = false,
+        kwargs...)
+    local x, cur, state
+
+    manifold = haskey(prob.kwargs, :manifold) ? prob.kwargs[:manifold] : nothing
+
+    if manifold === nothing || manifold !== opt.M
+        throw(ArgumentError("Either manifold not specified in the problem `OptimizationProblem(f, x, p; manifold = SymmetricPositiveDefinite(5))` or it doesn't match the manifold specified in the optimizer `$(opt.M)`"))
+    end
+
+    if data !== Optimization.DEFAULT_DATA
+        maxiters = length(data)
+    end
+
+    cur, state = iterate(data)
+
+    stopping_criterion = nothing
+    if maxiters !== nothing
+        stopping_criterion = StopAfterIteration(maxiters)
+    end
+
+    maxiters = Optimization._check_and_convert_maxiters(maxiters)
+    maxtime = Optimization._check_and_convert_maxtime(maxtime)
+
+    f = Optimization.instantiate_function(prob.f, prob.u0, prob.f.adtype, prob.p)
+
+    _loss = build_loss(f, prob)
+
+    gradF = build_gradF(f, prob, cur)
+
+    opt_res, opt_ret = call_manopt_optimizer(opt, _loss, gradF, prob.u0, stopping_criterion)
```

Comment:
I would change this interface for example to 
```suggestion
    opt_res, opt_ret = call_manopt_optimizer(opt, prob.M, _loss, gradF, prob.u0, stopping_criterion)
```
remove the manifold from all optimisers and be happy :)

Reply:
Oh yeah, good point. Will just remove it.